test/real_content/test_swupd.bats

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env bats
+
+load "../functional/testlib"
+
+# Test swupd with real official content published on clearlinux.org
+# This test assumes all functional tests to be passing in order to work,
+# because we aren't going to test all swupd functionalities. The idea is
+# to test major swupd commands and standard use cases
+
+#TODO: Add test for crossing a format barrier
+
+# Optional parameters
+# $URL: Clearlinux content and version urls
+# $MAX_VERSIONS: The maximum number of versions we are going to be looking to
+#                test updates. Should be # at least 2 (current + update).
+#                Default is 10.
+# $BUNDLE_LIST: List of bundles to be installed on the system besides os-core
+#               and os-core-update. If empty, install all bundles.
+
+local SWUPD_OPTS # Swupd options to use in tests
+local FORMAT # Format of the last available version
+local VERSIONS # Array of last available versions. First item is the oldest
+local ROOT_DIR # Root directory of the installation
+
+# get_format: Echos the format of the version in $1
+# TODO: This is not the best way to handle that. we should have in swupd
+# codebase the information about the supported format of this swupd.
+get_format() {
+	v="$1"
+	curl -f ${URL}/${v}/format 2>/dev/null
+	return 0
+}
+
+# get_format_from_versions: Echos the first format available in a version from
```

Comment:
one of the things I see you are often overlooking is using variables without enclosing them in double quotes.
> Quoting variables prevents word splitting and glob expansion, and prevents the script from breaking when input contains spaces, line feeds, glob characters and such. - https://github.com/koalaman/shellcheck/wiki/SC2086

>